include/matrix3x4SIMD_impl.h

```diff
@@ -210,16 +210,16 @@ inline vectorSIMDf matrix3x4SIMD::getTranslation3D() const
 
 inline matrix3x4SIMD& matrix3x4SIMD::setScale(const core::vectorSIMDf& _scale)
 {
-	const __m128i mask0001 = BUILD_MASKF(0, 0, 0, 1);
+	const vectorSIMDu32 mask0001 = vectorSIMDu32(BUILD_MASKF(0, 0, 0, 1));
 
 	const vectorSIMDu32& scaleAlias = reinterpret_cast<const vectorSIMDu32&>(_scale);
 
 	vectorSIMDu32& rowAlias0 = reinterpret_cast<vectorSIMDu32&>(rows[0]);
 	vectorSIMDu32& rowAlias1 = reinterpret_cast<vectorSIMDu32&>(rows[1]);
 	vectorSIMDu32& rowAlias2 = reinterpret_cast<vectorSIMDu32&>(rows[2]);
-	rowAlias0 = (scaleAlias & vectorSIMDf(BUILD_MASKF(1, 0, 0, 0))) | (rowAlias0 & vectorSIMDf(mask0001));
-	rowAlias1 = (scaleAlias & vectorSIMDf(BUILD_MASKF(0, 1, 0, 0))) | (rowAlias1 & vectorSIMDf(mask0001));
-	rowAlias2 = (scaleAlias & vectorSIMDf(BUILD_MASKF(0, 0, 1, 0))) | (rowAlias2 & vectorSIMDf(mask0001));
+	rowAlias0 = (scaleAlias & vectorSIMDf(vectorSIMDu32(BUILD_MASKF(1, 0, 0, 0)))) | (rowAlias0 & vectorSIMDf(mask0001));
+	rowAlias1 = (scaleAlias & vectorSIMDf(vectorSIMDu32(BUILD_MASKF(0, 1, 0, 0)))) | (rowAlias1 & vectorSIMDf(mask0001));
+	rowAlias2 = (scaleAlias & vectorSIMDf(vectorSIMDu32(BUILD_MASKF(0, 0, 1, 0)))) | (rowAlias2 & vectorSIMDf(mask0001));
```

Comment:
this should be a `reinterpret_Cast` because as it stands it will convert the integer values to floats (not reinterpret the bits)